ACISobs.py

```diff
@@ -790,6 +793,38 @@ def get_instrument(self, observation):
         instrument = observation[self.in_focal_plane]
         return(instrument)
 
+    #----------------------------------------------------------------------
+    #
+    # get_all_instruments
+    #
+    #---------------------------------------------------------------------
+    def get_all_instruments(self, observations):
+        """
+        Given a list element from the list of obsids extracted by this 
+        class, extract the instruments, append to a list  and return it
+        """
+        instrument_list = []
```

Comment:
This works, but the preferred idiom in Python is using list comprehension:
```
instrument_list = [self.get_instrument(obs) for obs in observations]
```
This is both faster and easier to read.